### PR TITLE
part of the e2e test of the kubeclipper

### PR DESCRIPTION
**Pull request by Metrora**
Signed-off-by: Liucw <liu.chuwei@99cloud.net>

### What type of PR is this?

/area testing

### Which issue(s) this PR fixes:

Fixes https://github.com/kubeclipper-labs/kubeclipper/issues/75

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```
@x893675

### CODE REVIEW

File: test/e2e/node/terminal.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	msg := ""
+	nodeID := ""
+	token := &oauth.Token{}
+	credential := &v1.SSHCredential{}
+
+	ginkgo.BeforeEach(func() {
+		ginkgo.By("Check that there are enough available nodes")
+		nodes, err := f.Client.ListNodes(context.TODO(), kc.Queries{
+			Pagination:    query.NoPagination(),
+			LabelSelector: fmt.Sprintf("!%s", common.LabelNodeRole),
```
Comment (lixd) [inline test/e2e/node/terminal.go:32]: used node can test terminal too,remove LabelSelector.

File: test/e2e/node/terminal.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		ginkgo.By("Get public key")
+		pub, err := f.Client.GetPublicKey(context.TODO())
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Get token")
+		body := kc.LoginRequest{
+			Username: "admin",
+			Password: "Thinkbig1",
+		}
+		token, err = f.Client.Login(context.TODO(), body)
```
Comment (lixd) [inline test/e2e/node/terminal.go:46]: kc.client add a  method to get token,like get host,refer pkg/simple/client/kc/client.go

File: test/e2e/node/disable_node.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	"github.com/kubeclipper/kubeclipper/pkg/simple/client/kc"
+	cluster2 "github.com/kubeclipper/kubeclipper/test/e2e/cluster"
+
+	"github.com/onsi/ginkgo"
+
+	"github.com/kubeclipper/kubeclipper/pkg/scheme/common"
+	"github.com/kubeclipper/kubeclipper/test/framework"
+)
+
+var _ = cluster2.SIGDescribe("[Slow] [Serial] Disable node", func() {
```
Comment (lixd) [inline test/e2e/node/disable_node.go:35]: disable node is fast, replace [Slow] tag.and check others tag.

File: test/e2e/node/disable_node.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		ginkgo.By("disable node")
+		err := f.Client.DisableNode(context.TODO(), nodeID)
+		framework.ExpectNoError(err)
+	})
+
+	ginkgo.BeforeEach(func() {
+		ginkgo.By("Check that there are enough available nodes")
+		nodes, err := f.Client.ListNodes(context.TODO(), kc.Queries{
+			Pagination:    query.NoPagination(),
+			LabelSelector: fmt.Sprintf("!%s", common.LabelNodeRole),
```
Comment (lixd) [inline test/e2e/node/disable_node.go:94]: maybe you want common.LabelNodeDisable

File: test/e2e/node/disable_node.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+		ginkgo.By("enable node")
+		err := f.Client.EnableNode(context.TODO(), nodeID)
+		framework.ExpectNoError(err)
+	})
+
+	ginkgo.BeforeEach(func() {
+		ginkgo.By("Check that there are enough available nodes")
+		nodes, err := f.Client.ListNodes(context.TODO(), kc.Queries{
+			Pagination:    query.NoPagination(),
+			LabelSelector: fmt.Sprintf("!%s", common.LabelNodeRole),
```
Comment (lixd) [inline test/e2e/node/disable_node.go:49]: maybe you want !common.LabelNodeDisable

File: test/e2e/cluster/upgrade.go [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+	ginkgo.BeforeEach(func() {
+		clus, err := createClusterBeforeEach(f, "cluster-aio", initOnlineAIOCluster)
+		framework.ExpectNoError(err)
+		clu = clus.Items[0].DeepCopy()
+	})
+
+	ginkgo.It("upgrade online cluster and ensure cluster is upgraded", func() {
+		ginkgo.By("upgrade cluster")
+		err := f.Client.UpgradeCluster(context.TODO(), clu.Name, initUpgradeCluster(false, "v1.23.9"))
```
Comment (lixd) [inline test/e2e/cluster/upgrade.go:37]: online upgrade needs set mirrors when creating cluster

File: test/e2e/cluster/upgrade.go [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+	ginkgo.BeforeEach(func() {
+		clus, err := createClusterBeforeEach(f, "cluster-aio", initAIOCluster)
+		framework.ExpectNoError(err)
+		clu = clus.Items[0].DeepCopy()
+	})
+
+	ginkgo.It("upgrade offline cluster and ensure cluster is upgraded", func() {
+		ginkgo.By("upgrade cluster")
+		err := f.Client.UpgradeCluster(context.TODO(), clu.Name, initUpgradeCluster(true, "v1.23.9"))
```
Comment (lixd) [inline test/e2e/cluster/upgrade.go:74]: offline upgrade needs check offline packages

File: pkg/simple/client/kc/client.go [hunk 8]
```diff
@@ -71,6 +71,11 @@ func (cli *Client) Host() string {
 	return cli.host
 }
 
+func (cli *Client) Token(ctx context.Context, body LoginRequest) (string, error) {
+	token, err := cli.Login(ctx, body)
```
Comment (lixd) [inline pkg/simple/client/kc/client.go:75]: just return cli.bearerToken,without login.

**Review by lixd — APPROVED** [review]
/lgtm

File: pkg/query/query.go [hunk 9]
```diff
@@ -84,6 +86,9 @@ type Query struct {
 	AllowWatchBookmarks  bool
 	ResourceVersionMatch string
 	FuzzySearch          map[string]string
+	Name                 string
```
Comment (x893675) [inline pkg/query/query.go:89]: rename

File: pkg/query/query.go [hunk 10]
```diff
@@ -84,6 +86,9 @@ type Query struct {
...
+	Token                string
```
Comment (x893675) [inline pkg/query/query.go:90]: need rename

File: pkg/query/query.go [hunk 11]
```diff
@@ -84,6 +86,9 @@ type Query struct {
...
+	Message              string
```
Comment (x893675) [inline pkg/query/query.go:91]: need rename

File: test/e2e/cluster/add_node.go [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+		framework.ExpectNoError(err)
+	})
+
+	ginkgo.BeforeEach(func() {
+		clus, err := createClusterBeforeEach(f, "cluster-aio", initAIOCluster)
+		framework.ExpectNoError(err)
+		clu = clus.Items[0].DeepCopy()
+	})
+
+	ginkgo.BeforeEach(func() {
```
Comment (Xvv-v) [inline test/e2e/cluster/add_node.go:39]: 为啥要写两个 beforeEach 啊

**Pull request merged**